WordPressCom-Stats-iOS/Services/WPStatsServiceRemote.m

```diff
@@ -182,6 +185,16 @@ - (void)batchFetchInsightsStatsWithAllTimeCompletionHandler:(StatsRemoteAllTimeC
     if (publicizeCompletion) {
         [mutableOperations addObject:[self operationForPublicizeWithCompletionHandler:publicizeCompletion]];
     }
+    if (streakCompletion) {
+        unsigned unitFlags = NSCalendarUnitYear | NSCalendarUnitMonth |  NSCalendarUnitDay;
+        NSDate *now = [NSDate date];
+        NSCalendar *gregorian = [NSCalendar currentCalendar];
```

Comment:
Its probably safer to pull the Gregorian calendar because the current one may be non-Gregorian. I think there are a couple other places I have to fix this in the project.

```objc
[[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
```